app/core/store/modules/campaigns.js

```diff
@@ -0,0 +1,37 @@
+import campaignsApi from 'core/api/campaigns'
+
+export default {
+  namespaced: true,
+  state: {
+    byId: {},
+    bySlug: {}
+  },
+
+  mutations: {
+    setCampaignData: (state, campaignData) => {
+      Vue.set(state.byId, campaignData._id, campaignData)
+      Vue.set(state.bySlug, campaignData.slug, campaignData)
+    }
+  },
+
+  getters: {
+    getCampaignData: (state) => (idOrSlug) => {
+      return state.byId[idOrSlug] || state.bySlug[idOrSlug]
+    }
+  },
+
+  actions: {
+    fetch: async ({ commit, state }, campaignHandle) => {
+      if (state.byId[campaignHandle] || state.bySlug[campaignHandle]) {
+        return
+      }
+      try {
+        const campaignData = await campaignsApi.get({ campaignHandle: campaignHandle })
+        commit('setCampaignData', campaignData)
+      } catch (e) {
+        console.error('Error in fetching campaign', e)
+        noty({ text: 'Fetch campaign failure', type: 'error' })
```

Comment:
Can we add `TODO error handling` comments to these noty lines so that we know to go back and fix them later?  Adding a consistent comment allows us to grep for them once we decide on a strategy.

Reply:
@jmif This seems useful for us to adopt.